src/jit/block.h

```diff
@@ -144,6 +144,85 @@ struct EntryState
     StackEntry* esStack;               // ptr to  stack
 };
 
+// Enumeration of the kinds of memory whose state changes the compiler tracks
+enum MemoryKind
+{
+    ByrefExposed = 0, // Includes anything byrefs can read/write (everything in GcHeap, address-taken locals,
+                      //                                          unmanaged heap, callers' locals, etc.)
+    GcHeap,           // Includes actual GC heap, and also static fields
+    MemoryKindCount,  // Number of MemoryKinds
+};
+#ifdef DEBUG
+const char* const memoryKindNames[] = {"ByrefExposed", "GcHeap"};
+#endif // DEBUG
+
+// Bitmask describing a set of memory kinds (usable in bitfields)
+typedef unsigned int MemoryKindSet;
+
+// Bitmask for an empty MemoryKindSet
+inline MemoryKindSet memoryKindSet()
+{
+    return 0;
+}
+
+// Bitmask for a MemoryKindSet containing the specified MemoryKinds
+template <typename... MemoryKinds>
+inline MemoryKindSet memoryKindSet(MemoryKind id, MemoryKinds... ids)
+{
+    return (1U << id) | memoryKindSet(ids...);
+}
```

Comment:
It's called a variadic template, and yes it's supported across our compilers (and already used in a  [couple](https://github.com/dotnet/coreclr/blob/93cb39e3c1bbd4407261926a7365949f288ebc37/src/jit/jitstd/list.h#L476-L480) other [places](https://github.com/dotnet/coreclr/blob/93cb39e3c1bbd4407261926a7365949f288ebc37/src/jit/gentree.h#L1085-L1088) in the jit)